File: app/src/main/java/com/lebedevsd/currencyrates/utils/CountryFlag.java
package com.lebedevsd.currencyrates.utils;

import android.text.TextUtils;

import java.util.Locale;

public class CountryFlag {

    public static String flag(Locale locale) {
        return CountryFlag.flag(locale.getCountry().toLowerCase());
    }

    public static String flag(String iso) {
        if (TextUtils.isEmpty(iso)) {
            return "_unknown";
        }
        if (iso.length() > 3) {
            iso = iso.substring(0, 3);
        }
        return "flag_" + iso.toLowerCase();
    }
}

File: app/src/main/java/com/lebedevsd/currencyrates/utils/CountryFlag.kt
package com.lebedevsd.currencyrates.utils

import android.annotation.SuppressLint
import android.text.TextUtils

object CountryFlag {

    @SuppressLint("DefaultLocale")
    fun flag(iso: String): String {
        if (TextUtils.isEmpty(iso)) {
            return "_unknown"
        }

        return "flag_" + (if (iso.length > 3) {
            iso.substring(0, 3)
        } else iso).toLowerCase()
    }
}
